src/include/mpir_request.h

```diff
@@ -409,6 +409,7 @@ MPL_STATIC_INLINE_PREFIX MPIR_Request *MPIR_Request_create_null_recv(void)
 static inline void MPIR_Request_free_with_safety(MPIR_Request * req, int need_safety)
 {
     int inuse;
+    int pool = (req->handle & REQUEST_POOL_MASK) >> REQUEST_POOL_SHIFT;
```

Comment:
Use `get_vci_wrapper`?

Reply:
`get_vci_wrapper` is internal to vci. We still maintain vci as a ch4-only concept. Maybe we should just make it into a global concept since there are so many places we have leaked it to the upper-layer already? Anyway, let me create a `MPIR_REQUEST_POOL` macro, and refactor the ch4 functions to use that.

Reply:
Oh, `get_vci_wrapper` is for special treatment of persistent request in progress. It is not needed in `MPIR_Request_free`.

Reply:
@raffenet Added a commit to wrap the direct accessing of request handle bit patterns. Could you review it and also see whether I addressed your comment.

Reply:
I'm also thinking use of a wrapper would be better here compared to the expression: `(req->handle & REQUEST_POOL_MASK) >> REQUEST_POOL_SHIFT`

Reply:
@sagarth Could you check if my last commit addresses your comment?

Reply:
Yes it does. Thanks.